sys/include/net/gnrc/mac/internal.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_mac  A common MAC type for providing key MAC parameters and helper functions
+ * @ingroup     net
+ * @brief       A common MAC type for providing key MAC parameters and helper functions.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for internal functions of MAC protocol
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_MAC_INTERNAL_H_
+#define GNRC_MAC_INTERNAL_H_
+
+#include <stdint.h>
+#include <net/ieee802154.h>
+#include <net/gnrc/mac/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* @brief Extract the destination address out of an GNRC_NETTYPE_NETIF pktsnip
+ *
+ * @param[in]   pkt                 pktsnip from whom to extract
+ * @param[out]  pointer_to_addr     pointer to address will be stored here
+ *
+ * @return                          length of destination address
+ */
+int gnrc_mac_get_dstaddr(gnrc_pktsnip_t* pkt, uint8_t* pointer_to_addr[]);
+
+/* @brief Find the first pktsnip of @p type
+ *
+ * Will search linearly through the packet buffer @p pkt and yield
+ * gnrc_pktsnip_t::data of the first pktsnip match the type @p type.
+ *
+ * @param[in]   pkt     pktsnip that will be searched
+ * @param[in]   type    type to search for
+ *
+ * @return              pointer to data, NULL is not found
+ */
+void* gnrc_mac_pktbuf_find(gnrc_pktsnip_t* pkt, gnrc_nettype_t type);
+
+/* @brief Check if packet is for broadcast
+ *
+ * @param[in]   pkt     packet to check
+ *
+ * @return              true if the packet is for broadcast, otherwise false
+ */
+static inline bool gnrc_mac_chk_pkt_bcast(gnrc_pktsnip_t* pkt)
```

Comment:
It seems a bit arbitrary to put a `gnrc_netif_hdr` specific function here. I would rather put it in that header.